pegasus/sites.v3/hourofcode.com/public/js/signup-form.js

```diff
@@ -0,0 +1,168 @@
+      // out of school, either US or non-US
+      $('#organization-name-field').show();
+      $('#hoc-event-location-field').show();
+      $('#school-name-field').hide();
+      $('#hoc-entire-school').hide();
+      $('.continue-btn').hide();
+      $('#submit').show();
+    }
+  }
+
+  $('#country').change(function () {
+    checkShowNameEventLocation();
+  });
+
+  $('#hoc-event-type').change(function () {
+    checkShowNameEventLocation();
+  });
+
+  function checkShowCensusFollowUp() {
+    if ($("#twenty-hour-how-much").val() === "some" || $("#twenty-hour-how-much").val() === "all" || $("#ten-hour-how-much").val() === "some" ||
+    $("#ten-hour-how-much").val() === "all") {
+      $('#followup_questions').show();
+    } else {
+      $('#followup_questions').hide();
+    }
+  }
+
+  $('#twenty-hour-how-much').change(function () {
+    checkShowCensusFollowUp();
+  });
+
+  $('#ten-hour-how-much').change(function () {
+    checkShowCensusFollowUp();
+  });
+
+  $('#role').change(function () {
+    if ($(this).val() === "teacher" || $(this).val() === "administrator") {
+      $('#pledge').show();
+    } else {
+      $('#pledge').hide();
+    }
+  });
+ });
+
+function signupFormComplete(data) {
+  window.location = "#{resolve_url('thanks')}";
+}
+
+function validateFields() {
+  if ($("#hoc-name").val() === "") {
+    $('#name-error').show();
+    return false;
+  } else {
+    $('#name-error').hide();
+  }
+
+  if ($("#hoc-email").val() === "") {
+    $('#email-error').show();
+    return false;
+  } else {
+    $('#email-error').hide();
+  }
+
+  if ($("#country").val() === "") {
+    $('#country-error').show();
+    return false;
+  } else {
+    $('#country-error').hide();
+  }
+
+  if ($("#hoc-event-type").val() === "") {
+    $('#event-type-error').show();
+    return false;
+  } else {
+    $('#event-type-error').hide();
+  }
+
+  if  ($("#hoc-event-type").val() === "in_school") {
+    if ($("#school-name").val() === "") {
+      $('#school-name-error').show();
+      return false;
+    } else {
+      $('#school-name-error').hide();
+    }
+  }
+
+  if  ($("#hoc-event-type").val() === "out_of_school") {
+    if ($("#organization-name").val() === "") {
+      $('#organization-name-error').show();
+      return false;
+    } else {
+      $('#organization-name-error').hide();
+    }
+  }
+
+  if ($("#hoc-event-location").val() === "") {
+    $('#event-location-error').show();
+    return false;
+  } else {
+    $('#event-location-error').hide();
+  }
+  return true;
+}
+
+function signupFormError(data) {
+  $('#error_message').html("<p>#{signup_submit_error_message}</p>").show();
+  $("#signup_submit").removeAttr('disabled');
+}
+
+function signupFormSubmit() {
+  $("#signup_submit").attr('disabled','disabled');
+
+  $.ajax({
+    url: "/forms/HocSignup#{hoc_year}",
+    type: "post",
+    dataType: "json",
+    data: $("#hoc-signup-form").serialize()
+  }).done(signupFormComplete).fail(signupFormError);
+}
```

Comment:
As a followup, can we put this js in apps so it gets linting (and could have unit tests)?